### ISSUE TITLE
ERROR failed to acquire resources on one or more segments  (entry db x.x.x.x) for specific SQL model version: 6X_STABLE

### ISSUE THREAD

**Issue by xuangl** [greenplum-db/gpdb#9910]
#### 1, pg_hba.conf
```
local	all	gpadmin	ident
host	all	gpadmin	127.0.0.1/28	trust
host	all	gpadmin	12.0.217.98/32	trust
local	replication	gpadmin	ident
host	replication	gpadmin	samenet	trust
host	all	all	0.0.0.0/0	md5
local	gpperfmon	gpmon	md5
host	all	gpmon	127.0.0.1/28	md5
host	all	gpmon	::1/128	md5
host	all	gpmon	samenet	md5
```

#### 2,  initialization
psql -c "CREATE ROLE gptest WITH LOGIN PASSWORD 'gptest11#'"
createdb -O gptest ods
psql -d ods -c "CREATE SCHEMA ods"
psql -d ods -c "ALTER SCHEMA ods OWNER TO gptest"

#### 3，Specific SQL mode for error reporting
log in to GP with username and password（non gpadmin）
select count(1) from B where time < (select max(time) from A)

<img width="630" alt="屏幕快照 2020-04-15 16 46 52" src="https://user-images.githubusercontent.com/1785959/79334985-d7566d00-7f53-11ea-8512-01f5b5715cb9.png">

#### 4，error log

```2020-04-13 19:02:54.598266 CST,"gptest","gptest",p17962,th193386624,"11.0.114.111","53408",2020-04-13 18:53:57 CST,0,con22,cmd10,seg-1,,dx339,,sx1,"LOG","00000","statement: select count(1) from t_dt where issue_time < (select max(issue_time) from t_dt)",,,,,,"select count(1) from t_dt where issue_time < (select max(issue_time) from t_dt)",0,,"postgres.c",1617,
2020-04-13 19:02:54.694615 CST,"gptest","gptest",p17962,th193386624,"11.0.114.111","53408",2020-04-13 18:53:57 CST,0,con22,cmd10,seg-1,,dx339,,sx1,"DEBUG1","00000","GPORCA produced plan",,,,,,"select count(1) from t_dt where issue_time < (select max(issue_time) from t_dt)",0,,"orca.c",51,
2020-04-13 19:02:54.878764 CST,"gptest","gptest",p19064,th193386624,"127.0.0.1","38858",2020-04-13 19:02:54 CST,0,con22,,seg-1,,,,,"DEBUG1","00000","InitProcess(): gp_session_id 22, Gp_role 2",,,,,,,0,,"proc.c",478,
2020-04-13 19:02:54.988550 CST,"gptest","gptest",p19064,th193386624,"127.0.0.1","38858",2020-04-13 19:02:54 CST,0,con22,,seg-1,,,,,"DEBUG1","00000","SessionState_Acquire: pinCount: 2, activeProcessCount: 1",,,,,,,0,,"session_state.c",120,"Stack trace:
2020-04-13 19:02:54.993995 CST,"gptest","gptest",p19064,th193386624,"127.0.0.1","38858",2020-04-13 19:02:54 CST,0,con22,,seg-1,,,,sx1,"LOG","00000","rejecting TCP connection to master using internalconnection protocol",,,,,,,0,,"auth.c",361,
2020-04-13 19:02:54.994804 CST,"gptest","gptest",p19064,th193386624,"127.0.0.1","38858",2020-04-13 19:02:54 CST,0,con22,,seg-1,,,,sx1,"DEBUG1","00000","Found writer proc entry.  My Pid 19064, his pid 17962",,,,,,,0,,"lock.c",856,
2020-04-13 19:02:54.994863 CST,"gptest","gptest",p19064,th193386624,"127.0.0.1","38858",2020-04-13 19:02:54 CST,0,con22,,seg-1,,,,sx1,"DEBUG1","00000","Reader found lock AccessShareLock on object 0/1260/0 doesn't conflict",,,,,,,0,,"lock.c",1077,
2020-04-13 19:02:54.994899 CST,"gptest","gptest",p19064,th193386624,"127.0.0.1","38858",2020-04-13 19:02:54 CST,0,con22,,seg-1,,,,sx1,"DEBUG1","00000","Reader found lock AccessShareLock on object 0/1260/0 doesn't conflict",,,,,,,0,,"lock.c",1077,
2020-04-13 19:02:54.994921 CST,"gptest","gptest",p19064,th193386624,"127.0.0.1","38858",2020-04-13 19:02:54 CST,0,con22,,seg-1,,,,sx1,"DEBUG1","00000","Reader found lock AccessShareLock on object 0/2676/0 doesn't conflict",,,,,,,0,,"lock.c",1077,
2020-04-13 19:02:55.045472 CST,"gptest","gptest",p17962,th193386624,"11.0.114.111","53408",2020-04-13 18:53:57 CST,0,con22,cmd11,seg-1,,dx339,,sx1,"ERROR","58M01","failed to acquire resources on one or more segments","fe_sendauth: no password supplied
2020-04-13 19:02:55.045612 CST,"gptest","gptest",p17962,th193386624,"11.0.114.111","53408",2020-04-13 18:53:57 CST,0,con22,cmd11,seg-1,,dx339,,sx1,"LOG","00000","An exception was encountered during the execution of statement: select count(1) from t_dt where issue_time < (select max(issue_time) from t_dt)",,,,,,,0,,,,
```

#### 5, can fix @gaos1 
`gpconfig -c gp_reject_internal_tcp_connection -v off --skipvalidation`

or

``` pg_hba.conf 
host all all 127.0.0.1/28 trust
host all all 0.0.0.0/0 md5
```

or

```
set optimizer=off;
```

**Comment by xuangl** [comment]
@adam8157

**Comment by gaos1** [comment]
the pg_hba.conf content is added by you, or created by installer/initsystem?
```
host all all 0.0.0.0/0 md5
```

**Comment by xuangl** [comment]
> the pg_hba.conf content is added by you, or created by installer/initsystem?
> 
> ```
> host all all 0.0.0.0/0 md5
> ```

I added it manually and then I ran the command gpstop -u

**Comment by gaos1** [comment]
why do you add them? what is the problem you hit before add them?

**Comment by xuangl** [comment]
> why do you add them? what is the problem you hit before add them?

I hope users can log in remotely through user name and password so added host all all 0.0.0.0/0 md5.

Users newly added before adding cannot log in with password, whether remote or local.

The user I created with the following command

psql -c "CREATE ROLE gptest WITH LOGIN PASSWORD 'gptest'"

**Comment by gaos1** [comment]
in this case, you only need to add it in master pg_hba. did you add the same to segments? what is your segment pg_hba ?

**Comment by xuangl** [comment]
> in this case, you only need to add it in master pg_hba. did you add the same to segments? what is your segment pg_hba ?

master 12.0.217.98
segment hosts 12.0.217.61,12.0.217.62,12.0.217.63

I didn't change the segment  configuration.

host all all 12.0.217.98/32 trust
host all gpadmin 12.0.217.61/32 trust
host  replication gpadmin samenet trust
host all gpadmin 12.0.217.62/32 trust
host  replication gpadmin samenet trust
host all gpadmin 12.0.217.62/32 trust

**Comment by xuangl** [comment]
@gaos1

**Comment by xuangl** [comment]
set optimizer=off;
explain select count(1) from test_issue where age<(select max(age) from test_issue);
<img width="568" alt="屏幕快照 2020-04-15 21 08 54" src="https://user-images.githubusercontent.com/1785959/79340826-7764c400-7f5d-11ea-802e-20751b2d258b.png">

set optimizer=on;
explain select count(1) from test_issue where age<(select max(age) from test_issue);
<img width="565" alt="屏幕快照 2020-04-15 21 08 38" src="https://user-images.githubusercontent.com/1785959/79340886-85b2e000-7f5d-11ea-80b6-ddb1975f82ca.png">

**Comment by xuangl** [comment]
single server, master and segment are deployed together, no error is reported.

**Comment by gaos1** [comment]
i don't have enough info, but I guess the trouble is from your change on master pg_hba.conf:
```
host all all 0.0.0.0/0 md5
```
This probably also rejected internal request if there is a entry db process on master.  you need to make sure all internal process are accessible via trust.

**Comment by xuangl** [comment]
> i don't have enough info, but I guess the trouble is from your change on master pg_hba.conf:
> 
> ```
> host all all 0.0.0.0/0 md5
> ```
> 
> This probably also rejected internal request if there is a entry db process on master. you need to make sure all internal process are accessible via trust.

Then I should configure it to allow users to access the GP remotely ?

**Comment by xuangl** [comment]
I found a problem, as long as it is a cross server deployment segment, it will report an error

@gaos1

**Comment by xuangl** [comment]
> > i don't have enough info, but I guess the trouble is from your change on master pg_hba.conf:
> > ```
> > host all all 0.0.0.0/0 md5
> > ```
> > 
> > 
> > This probably also rejected internal request if there is a entry db process on master. you need to make sure all internal process are accessible via trust.
> 
> Then I should configure it to allow users to access the GP remotely ?

Now it's like this. A certain step of the execution plan reports this error
```
ERROR:  failed to acquire resources on one or more segments
DETAIL:  FATAL:  no pg_hba.conf entry for host "127.0.0.1", user "gptest", database "ogg", SSL off
 (entry db x.x.x.x:5432)
, Time: 0.142000s
```

I changed the configuration, but I still report an error
```
host all gptest 0.0.0.0/0 md5
```

**Comment by xuangl** [comment]
> > i don't have enough info, but I guess the trouble is from your change on master pg_hba.conf:
> > ```
> > host all all 0.0.0.0/0 md5
> > ```
> > 
> > 
> > This probably also rejected internal request if there is a entry db process on master. you need to make sure all internal process are accessible via trust.
> 
> Then I should configure it to allow users to access the GP remotely ?

Error reported when executing SQL in remote mode
```
ERROR:  failed to acquire resources on one or more segments
DETAIL:  FATAL:  no pg_hba.conf entry for host "127.0.0.1", user "gptest", database "ogg", SSL off
 (entry db x.x.x.:5432)
, Time: 0.142000s
```

But I didn't configure it 
```
"127.0.0.1", user "gptest", database "ogg"
```

I think this is the internal execution of GP and should not be specially configured by the administrator.

Now every time I add a remote user, I will configure remote and local access to avoid error reporting

```
host all  ${user}  127.0.0.1/28  trust
host all  ${user}  0.0.0.0/0 md5
```

**Comment by xuangl** [comment]
This problem does not exist on gp5.26

**Comment by xuangl** [comment]
Previously, I configured PGHOST = 127.0.0.1 in .bashrc. After commenting it out, restarting GP will not report an error. I found this description in GP source code. I don't know if it matters
![屏幕快照 2020-04-18 11 59 54](https://user-images.githubusercontent.com/1785959/79627582-27a81780-816c-11ea-8438-253f052d1589.png)

I tested that this problem does not exist in gp5.26.

The process of reproducing problems in GP6

Greenplum version 6.1.0
OS version Linux 3.10.0-693.el7.x86_64
Cluster mode, deploying segments across servers
Configure export PGHOST=127.0.0.1 to .bashrc. make it effective
start cluster
Specific SQL:   select count(1) from A where c < (select max(c) from A)
